functions/Get-DbaTable.ps1

```diff
@@ -0,0 +1,176 @@
+
+#>
+	[CmdletBinding()]
+	param ([parameter(ValueFromPipeline, Mandatory = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[object[]]$SqlInstance,
+		[System.Management.Automation.PSCredential]$SqlCredential,
+		[switch]$IncludeSystemDBs,
+		[switch]$Silent,
+		[string[]]$Table
+	)
+	
+	DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SourceSqlCredential } }
+	
+	BEGIN
+	{
+		$databases = $psboundparameters.Databases
+		$exclude = $psboundparameters.Exclude
+		
+		$fqtns = @()
+		
+		if ($Table)
+		{
+			foreach ($t in $Table)
+			{
+				$dotcount = ([regex]::Matches($t, "\.")).count
+				
+				if ($dotcount -eq 1)
+				{
+					$schema = $t.Split(".")[0]
+					$table = $t.Split(".")[1]
+				}
+				
+				if ($dotcount -eq 2)
+				{
+					$database = $t.Split(".")[0]
+					$schema = $t.Split(".")[1]
+					$table = $t.Split(".")[2]
+				}
+				
+				$fqtn = [PSCustomObject] @{
+					Database = $database
+					Schema = $Schema
+					Table = $table
+				}
+				$fqtns += $fqtn
+			}
+		}
+	}
+	
+	PROCESS
+	{
+		foreach ($instance in $sqlinstance)
+		{
+			#For each SQL Server in collection, connect and get SMO object
+			Write-Message -Level Verbose -Message "Connecting to $instance"
+			
+			try
+			{
+				Write-Message -Level Verbose -Message "Connecting to $instance"
+				$server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
+			}
+			catch
+			{
+				Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance
+			}
+			
+			#If IncludeSystemDBs is true, include systemdbs
+			#only look at online databases (Status equal normal)
+			try
+			{
+				if ($databases.length -gt 0)
+				{
+					$dbs = $server.Databases | Where-Object { $databases -contains $_.Name }
+				}
+				elseif ($IncludeSystemDBs)
+				{
+					$dbs = $server.Databases | Where-Object { $_.status -eq 'Normal' }
+				}
+				else
+				{
+					$dbs = $server.Databases | Where-Object { $_.status -eq 'Normal' -and $_.IsSystemObject -eq 0 }
+				}
+				
+				if ($exclude.length -gt 0)
+				{
+					$dbs = $dbs | Where-Object { $exclude -notcontains $_.Name }
+				}
+			}
+			catch
+			{
+				Stop-Function -Message "Unable to gather dbs for $instance" -Target $instance -Continue
+			}
+			
+			foreach ($db in $dbs)
+			{
+				Write-Message -Level Verbose -Message "Processing $db"
+				
+				$d = $server.Databases[$db]
+				if ($fqtns)
+				{
+					foreach ($fqtn in $fqtns)
+					{
+						if ($fqtn.schema)
+						{
+							try
+							{
+								$db.Tables | Where-Object { $_.name -eq $table -and $_.Schema -eq $schema } | Select-DefaultView -Property Parent, Schema, Name, IndexSpaceUsed, DataSpaceUsed, RowCount, HasClusteredIndex, IsFileTable, IsMemoryOptimized, IsPartitioned, FullTextIndex, ChangeTrackingEnabled
+							}
+							catch
+							{
+								Write-Message -Level Warning "Could not find table name: $($fqtn.table) schema: $($fqtn.schema)"
```

Comment:
Please avoid positional binding, specify `-Message "Could ..."` instead.
When catching a non-critical error (one where you do not interrupt execution with Stop-Function for), you can still log the error to help troubleshooting. To do so, add `-ErrorRecord $_` as a parameter to Write-Message (nly when inside a catch block)